### PR TITLE
Cargo Transport Example

### PR DESCRIPTION
**Pull request by FarzanehMG**

### CODE REVIEW

File: CargoTransportService/AirTransport.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿namespace CargoTransportService;
+
+public class AirTransport : CargoTransport
+{
+	private static AirTransport _instance;
```
Comment (mohammadKarimi) [inline CargoTransportService/AirTransport.cs:5]: where you create an instance ? 
it is null here. and you are returning null object.

File: CargoTransportService/AirTransport.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	private AirTransport()
```
Comment (mohammadKarimi) [inline CargoTransportService/AirTransport.cs:7]: good one.

File: CargoTransportService/CargoTransport.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace CargoTransportService;
+
+public abstract class CargoTransport
```
Comment (mohammadKarimi) [inline CargoTransportService/CargoTransport.cs:3]: it is fully abstract and you can use interface instead. if you want.

File: CargoTransportService/TrainTransport.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿namespace CargoTransportService;
+
+public class TrainTransport : CargoTransport
+{
+	public TrainTransport()
+	{
+		Initialize();
+	}
+
+	private static void Initialize()
```
Comment (mohammadKarimi) [inline CargoTransportService/TrainTransport.cs:10]: why you make it static. please explain it.

File: CargoTransportService/TrainTransport.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	{
+		Console.WriteLine("Train transport initialized.");
+	}
+
+	public override void Book()
```
Comment (mohammadKarimi) [inline CargoTransportService/TrainTransport.cs:15]: public method must be appear before private.

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]

**Comment by FarzanehMG** [comment]
@mohammadKarimi I fixed the points, please review.

**Pull request closed without merging**